go-controller/cmd/ovnkube-trace/ovnkube-trace.go

```diff
@@ -364,72 +366,90 @@ func getSvcInfo(coreclient *corev1client.CoreV1Client, restconfig *rest.Config,
 // Modifies the svcInfo object the pointer of which is passed to it.
-func extractSubsetInfo(coreclient *corev1client.CoreV1Client, restconfig *rest.Config, subsets []corev1.EndpointSubset, svcInfo *SvcInfo, ovnNamespace, addressFamily string) error {
-	for _, subset := range subsets {
-		klog.V(5).Infof("==> Trying to extract information for service %s in namespace %s from subset %v",
-			svcInfo.SvcName, svcInfo.SvcNamespace, subset)
+// slice is *discoveryv1.EndpointSlice slices is
+func extractEndpointSliceInfo(coreclient *corev1client.CoreV1Client, restconfig *rest.Config, slices []discoveryv1.EndpointSlice, svcInfo *SvcInfo, ovnNamespace, addressFamily string) error {
+
+	for _, slice := range slices {
+		klog.V(5).Infof("==> Trying to extract information for service %s in namespace %s from slice %v",
+			svcInfo.SvcName, svcInfo.SvcNamespace, slice)
 
-		// Find a port for the subset.
+		// Find a port for the endpoint slice.
 		var podPort string
-		for _, port := range subset.Ports {
-			podPort = strconv.Itoa(int(port.Port))
+		for _, epPort := range slice.Ports {
+			if epPort.Port == nil {
+				klog.V(5).Infof("==> Endpoint Port is nil, skipping.")
+				continue // with the next port
+			}
+			podPort = strconv.Itoa(int(*epPort.Port))
 			if podPort == "" {
-				klog.V(5).Infof("==> Could not parse port %v, skipping.", port)
+				klog.V(5).Infof("==> Could not parse port %v, skipping.", epPort)
 				continue // with the next port
 			}
 		}
-		// Continue with the next subset if podPort is empty.
+		// Continue with the next slice if podPort is empty.
 		if podPort == "" {
-			continue // with the next subset
+			continue // with the next slice
 		}
 
-		// Parse pod information from the subset addresses. One of the subsets must have a valid address which does not belong
+		// Parse pod information from the endpoint slice addresses. One of the slices must have a valid address which does not belong
 		// to a host networked pod.
-		for _, epAddress := range subset.Addresses {
+		for _, endpoint := range slice.Endpoints {
+			epAddress := endpoint.Addresses
+			if len(epAddress) == 0 {
+				klog.V(5).Infof("Address is empty for endpoint. Skipping.")
+				continue
+			}
 			// This is nil for host networked services.
-			if epAddress.TargetRef == nil {
+			if endpoint.TargetRef == nil {
 				klog.V(5).Infof("Address %v belongs to a host networked pod. Skipping.", epAddress)
 				continue // with the next address
 			}
-			if epAddress.TargetRef.Name == "" || epAddress.TargetRef.Namespace == "" || epAddress.IP == "" {
+			if endpoint.TargetRef.Name == "" || endpoint.TargetRef.Namespace == "" {
 				klog.V(5).Infof("Address %v contains invalid data. podName %s, podNamespace %s, podIP %s. Skipping.",
-					epAddress, epAddress.TargetRef.Name, epAddress.TargetRef.Namespace, epAddress.IP)
+					epAddress, endpoint.TargetRef.Name, endpoint.TargetRef.Namespace, epAddress)
```

Comment:
`epAddresss` is printed twice here. the first argument is same as the last argument? do we need to? we can remove the last podIP I guess.